src/amo/reducers/home.js

```diff
@@ -145,6 +145,7 @@ export type ExternalResultShelfType = {|
   title: LocalizedString,
   url: string,
   endpoint: string,
+  addonType: number,
```

Comment:
As discussed, this should be a string, and in fact we have an `AddonTypeType` which can be used here.